lib/internal/streams/destroy.js

```diff
@@ -112,9 +113,7 @@ function _destroy(self, err, cb) {
       if (typeof then === 'function') {
         then.call(
           result,
-          function() {
-            process.nextTick(onDestroy, null);
-          },
+          nop,
```

Comment:
```suggestion
          function() {
            if (!called) {
              process.nextTick(onDestroy, null);
            }
          },
```

Reply:
As with https://github.com/nodejs/node/pull/41165/files#r768565452, I'd argue the `nop` is better at mimicking the non-async behaviour. If not, then `onDestroy` should at least be called with `err`, instead of `null`:

```js
if (!called) {
  process.nextTick(onDestroy, null);
}
```

Anyway, that was before knowing about #40773.